### PR TITLE
Update REST schemas for 2.3.5

### PR DESCRIPTION
**Pull request by keharper**
## Purpose of this pull request

This pull request (PR) updates the REST schemas for 2.3.5.

## Affected DevDocs pages

- https://devdocs.magento.com/redoc/2.3/admin-rest-api.html
- https://devdocs.magento.com/redoc/2.3/customer-rest-api.html
- https://devdocs.magento.com/redoc/2.3/guest-rest-api.html
- https://devdocs.magento.com/redoc/2.3/async-admin-rest-api.html
- https://devdocs.magento.com/redoc/2.3/async-customer-rest-api.html
- https://devdocs.magento.com/redoc/2.3/async-guest-rest-api.html

whatsnew
Updated the [REST API Reference](https://devdocs.magento.com/redoc/2.3/) for 2.3.5.

### CODE REVIEW

**Comment by keharper** [comment]
running tests

File: src/redoc/2.3/index.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ Each of the following links lead to a list of REST endpoints specific to a user
 
 ## Asynchronous endpoints
 
-You can run POST and PUT endpoints asynchronously while the `async.operations.all` message queue consumer is active. [Asynchronous web endpoints]({{page.baseurl}}/rest/asynchronous-web-endpoints.html) provides information about asynchronous routes, payloads, and responses.
+You can run POST, PUT, and DELETE endpoints asynchronously while the `async.operations.all` message queue consumer is active. [Asynchronous web endpoints]({{page.baseurl}}/rest/asynchronous-web-endpoints.html) provides information about asynchronous routes, payloads, and responses.
```
Comment (rogyar) [inline src/redoc/2.3/index.md:28]: Should we add PATCH request as well? The PATCH, POST, PUT, DELETE requests are supposed to be supported by async API. @nuzil might help us with this decision.

Comment (nuzil) [reply src/redoc/2.3/index.md:28]: Yes, @rogyar is correct. PATCH requests are also supported by Async API, despite there are no usage in the core of Magento, I would add this here as well, in respect to 3rd party integrtions

Comment (keharper) [reply src/redoc/2.3/index.md]: @rogyar @nuzil I added PATCH.

**Comment by keharper** [comment]
running tests

**Comment by keharper** [comment]
running tests

**Comment by keharper** [comment]
running tests

**Comment by keharper** [comment]
running tests

**Pull request merged**